autoload/vista/autocmd.vim

```diff
@@ -15,8 +15,7 @@ function! s:ClearOtherEvents(group) abort
 endfunction
 
 function! s:OnBufEnter(bufnr, fpath) abort
-  " If we just enter the buffer that is being displayed, skip the update.
-  if !exists('t:vista') || a:bufnr == t:vista.source.bufnr
+  if !exists('t:vista')
```

Comment:
It's ok to remove this _optimization_ as the auto update is always async.

Reply:
For this optimization to work, maybe `a:bufnr` shoud be checked against some "global source bufnr" variable? Because the __vista__ buffer is in fact global. I'm not familiar with the data structures of vista though, so I don't know the reason why the source bufnr should be kept as a `t:` variable.

Reply:
>the vista buffer is in fact global

That's the problem we should resolve. I thought it should be tab specific, not a global one. I usually use only one tab, so I actually did not observe this issue.